libsolutil/DominatorFinder.h

```diff
@@ -374,6 +380,41 @@ class DominatorFinder
 		}
 	}
 
+	/// Iteratively find the dominance frontier of each vertex.
+	/// The dominance frontier of a vertex ``w`` contains the first vertices reachable from ``w`` that
+	/// ``w`` does not dominate.
+	///
+	/// The algorithm iterates over the vertices in reverse DFS order, and for each vertex ``v`` with more than one predecessor ``w``,
+	/// it adds the vertex ``v`` to the dominance frontier of ``w`` if``w`` does not strictly dominates ``v``.
+	void findDominanceFrontier()
+	{
+		solAssert(!m_verticesInDFSOrder.empty());
+		solAssert(!m_immediateDominators.empty());
+
+		for (DfsIndex vIdx = m_verticesInDFSOrder.size()-1; vIdx > 0; --vIdx)
```

Comment:
I don't quite get why you need to traverse the vertices in reverse DFS order.
It seems to me this should work for any ordering of vertices.

Reply:
You could do that as `for (DfsIndex vIdx = 1; vIdx < m_verticesInDFSOrder.size(); ++vIdx)`, no?

Reply:
yeah, both works.